collections/src/main/scala/strawman/collection/immutable/SortedMap.scala

```diff
@@ -3,52 +3,126 @@ package collection
-        var result: CC[K, V2] = coll
-        val it = xs.iterator()
-        while (it.hasNext) result = result + it.next()
-        result
+    protected[this] def newSpecificBuilder(): Builder[K, SortedSet[K]] = sortedIterableFactory.newBuilder()
+
+    def rangeImpl(from: Option[K], until: Option[K]): SortedSet[K] = {
+      val map = self.rangeImpl(from, until)
+      new map.ImmutableKeySortedSet
     }
 
+    def empty: SortedSet[K] = sortedIterableFactory.empty
+
+    def incl(elem: K): SortedSet[K] = fromSpecificIterable(this).incl(elem)
+
+    def excl(elem: K): SortedSet[K] = fromSpecificIterable(this).excl(elem)
+  }
+
+  protected def mapFromIterable[K2, V2](it: collection.Iterable[(K2, V2)]): Map[K2, V2] =
+    Map.from(it)
+
+  // We override these methods to fix their return type (which would be `Map` otherwise)
+  def updated[V1 >: V](key: K, value: V1): CC[K, V1]
+
+  @`inline` final override def +[V1 >: V](kv: (K, V1)): CC[K, V1] = updated(kv._1, kv._2)
+
+  override def concat[V2 >: V](xs: collection.Iterable[(K, V2)]): CC[K, V2] = {
+    var result: CC[K, V2] = coll
+    val it = xs.iterator()
+    while (it.hasNext) result = result + it.next()
+    result
+  }
+
 }
 
-object SortedMap extends SortedMapFactory.Delegate[SortedMap](TreeMap)
\ No newline at end of file
+object SortedMap extends SortedMapFactory.Delegate[SortedMap](TreeMap) {
+
+  final class WithDefault[K, +V](underlying: SortedMap[K, V], d: K => V) extends SortedMap[K, V] {
+    override implicit def ordering: Ordering[K] = underlying.ordering
+
+    override def updated[V1 >: V](key: K, value: V1): SortedMap[K, V1] = new WithDefault[K, V1](underlying.updated[V1](key, value), d)
+
+    override def default(key: K): V = d(key)
+
+    override def mapFactory: MapFactory[Map] = underlying.mapFactory
+
+    override def get(key: K): Option[V] = underlying.get(key)
+
+    override def empty: SortedMap[K, V] = new WithDefault[K, V](underlying.empty, d)
+
+    override def iterator(): Iterator[(K, V)] = underlying.iterator()
+
+    override def rangeImpl(from: Option[K], until: Option[K]): SortedMap[K, V] =
+      new WithDefault[K, V](underlying.rangeImpl(from, until), d)
+
+    override def remove(key: K): SortedMap[K, V] = new WithDefault[K, V](underlying - key, d)
+
+    override def sortedMapFactory: SortedMapFactory[SortedMap] = underlying.sortedMapFactory
+
+    override protected[this] def sortedMapFromIterable[K2, V2](it: collection.Iterable[(K2, V2)])(implicit ordering: Ordering[K2]): SortedMap[K2, V2] =
+      SortedMap.from(it)
+
+    override def iteratorFrom(start: K): Iterator[(K, V)] = underlying.iteratorFrom(start)
+
+    override def keysIteratorFrom(start: K): Iterator[K] = underlying.keysIteratorFrom(start)
+
+    override protected[this] def fromSpecificIterable(coll: collection.Iterable[(K, V)]): SortedMap[K, V] =
+      new WithDefault[K, V](SortedMap.from(coll), d)
```

Comment:
Instead of using the “default” `SortedMap` factory you should use a factory method from `underlying`:

~~~ scala
new WithDefault[K, V](underlying.fromSpecificIterable(coll), d)
~~~

Reply:
Same problem as mentioned above `fromSpecificIterable` has access modifier of `protected[this]`. Maybe as a part of this fix, we can make it just `protected`. This will also make it easy to extract a common `Map.WithDefault[_]` type.